### PR TITLE
Sliding window maximum-->Closes: [#1750]

### PR DESCRIPTION
**Pull request by Prerna-07**
closes : #1750

### CODE REVIEW

**Comment by Prerna-07** [comment]
@nakul-19 @raksha009 @plazzy99 @tarun26091999 @rish-singhal @yasharth291 @Rohan27s @rudrakshi99 pls check

File: Array/readme.md [hunk 1]
```diff
@@ -33,4 +33,5 @@ To create an array, you need to known the length (or size) of the array in advan
 - Merging two sorted arrays in O(1) space ---->[C++](/Code/C++/merge_in_constant_space.cpp)
 - Peak element in array using linear search ----> [C++](/Code/C++/peak_value_linear_search.cpp) | [Java](/Code/Java/peakvalueinarray.java)
 - Repeating and Missing Number ----> [C++](/Code/C++/repeating_and_missing_number.cpp) | [Java](/Code/Java/Repeating_And_Missing_Number.java)
-- Union of two Arrays ----> [C++](Code/C++/Union_of_two_unsorted_array.cpp)
\ No newline at end of file
+- Union of two Arrays ----> [C++](Code/C++/Union_of_two_unsorted_array.cpp)
```
Comment (rish-singhal) [inline Array/readme.md:36]: sort this list in lexicographic order

File: Code/Java/Sliding Window Maximum.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+       18
+       8
+       12
+       3
+       0
+       5
+output: 19 19 19 19 19 17 32 32 32 32 32 18 18 18              
+*/
+/*
+Time Complexity: O(n)
```
Comment (rish-singhal) [inline Code/Java/Sliding Window Maximum.java:117]: describe n here

File: Code/Java/Sliding Window Maximum.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+*/
+/*
+Test Case 2:
+input: 18
+       7
+       9
+       3
+       8
+       19
+       7
```
Comment (rish-singhal) [inline Code/Java/Sliding Window Maximum.java:100]: make this in one line, same with above test case

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by Prerna-07** [comment]
@rish-singhal updated required changes pls check.

**Comment by Prerna-07** [comment]
@rish-singhal pls review it again i have updated the content.

**Comment by Prerna-07** [comment]
@nakul-19 @raksha009 @plazzy99 @tarun26091999 @rish-singhal @yasharth291 @Rohan27s @rudrakshi99 pls review.

**Comment by rish-singhal** [comment]
resolve conflicts

**Comment by Prerna-07** [comment]
@rish-singhal done pls check.

File: Code/Java/sliding_window_maximum.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    Scanner scn = new Scanner(System.in);
+    int n = scn.nextInt();
+    int[] a = new int[n];
+    for(int i = 0; i < n; i++){
+       a[i] = scn.nextInt();
+    }
+    int k = scn.nextInt();
+    int [] r =right(a);                   // calling maximum element on right function it will return the index of maximum elements on right
+    int j=0;                              // let first element be maximum element
+    for(int i=0;i<a.length-k+1;++i)       // sliding the window
```
Comment (rish-singhal) [inline Code/Java/sliding_window_maximum.java:23]: give spaces around =, >, <,..... in whole code

File: Code/Java/sliding_window_maximum.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+/*
+Test Case 2: 
+input: 18
+       7 9 3 8 19 7 17 6 14 4 32 0 6 18 8 12 3 0 
+       5
+output: 19 19 19 19 19 17 32 32 32 32 32 18 18 18              
+*/
+/*
+Time Complexity: O(n)  // where n is the size of input array
+Space Complexity: O(n)
```
Comment (rish-singhal) [inline Code/Java/sliding_window_maximum.java:85]: add extra empty line at the end of the file

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by Prerna-07** [comment]
@rish-singhal  done pls check

**Comment by Prerna-07** [comment]
@rish-singhal pls check this also

File: Code/Java/sliding_window_maximum.java (hunk 4 above)
Comment (rish-singhal) [reply Code/Java/sliding_window_maximum.java:23]: Update this

File: Code/Java/sliding_window_maximum.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    int k = scn.nextInt();
+    int [] r =right(a);                   // calling maximum element on right function it will return the index of maximum elements on right
+    int j=0;                              // let first element be maximum element
+    for(int i=0;i<a.length-k+1;++i)       // sliding the window
+    {
+        if(j<i)                           // if element is out of window
+        {
+            j=i;                          // set first element in window the maximum element
+        }
+        while(r[j]<i+k)                   // here j is the index of current maximum element and r[j] will give the index of next maximum element, while r[j] is in the window
```
Comment (rish-singhal) [inline Code/Java/sliding_window_maximum.java:29]: This is not updated, add spaces

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by rish-singhal** [comment]
Hey please check my comments about giving appropriate spaces around operators.

**Comment by Prerna-07** [comment]
@rish-singhal  done pls check.

**Pull request merged**